dashboard-frontend/src/pages/NewStudyPage/controllers/simple.ts

```diff
@@ -0,0 +1,39 @@
+import {
+  initialiseFieldState,
+  updateFieldState,
+  Event,
+  getFieldState,
+} from '../../../helpers/state';
+import { Conf } from '../../../types/conf';
+import { ConfBase, FieldState } from '../../../types/form';
+
+const simple = (
+  conf: ConfBase,
+  localFormData?: Conf,
+  event?: Event,
+  fieldState?: FieldState[]
+) => {
+  const type: string = 'confObject';
+
+  if (conf.type === type) {
+    if (!localFormData && !fieldState && !event) {
+      return initialiseFieldState(conf);
+    }
+
+    if (!localFormData && fieldState && event) {
+      return updateFieldState(fieldState, event);
+    }
+
+    if (localFormData && !fieldState && !event) {
+      return getFieldState(conf, localFormData);
```

Comment:
I think we had this conversation, so forgive me if I'm forgetting something, but I'm not convinced localFormData is needed here at all. Why can't fieldState contain everything, including current values? 

This refers to Form as well. I'm not sure why localFormData is needed as a concept. It seems to me we have `fieldState`, which contains the fields and their values. Then, we have a translation to "formData" type.

Reply:
So when we fetch the existing form data, where should we plug those values into the value prop of each field object?

Does it not make sense to do that in the `translateField` helper like this? 

```
return {
    id: field.name,
    name: field.name,
    type: type,
    component: component,
    label: field.label,
    helper_text: field.helper_text ?? field.helper_text,
    options: field.options?.map((option: ConfBase | any) =>
      option.title
        ? {
            name: createNameFor(option.title),
            label: option.title,
          }
        : option
    ),
    value: localFormData ? localFormData[field.name] : getInitialValue(field),
  };
```
For this to work though we would need `localFormData` as a concept. I see what you mean about `fieldState` being the only data store needed to hold the field values but I think i'm missing something obvious here!